@stellar/anchor-tests/src/tests/sep31/transactions.ts

```diff
@@ -97,6 +112,97 @@ const canCreateTransaction: Test = {
       // but to satisfy TypeScript we make these checks here.
       throw "improperly configured";
     }
+
+    async function pollCustomerAccepted(
+      getCustomerNetworkCall: NetworkCall,
+      timeout: number,
+    ) {
+      for (let i = 0; i < timeout; i++) {
+        const resBody = await makeRequest(
+          getCustomerNetworkCall,
+          200,
+          result,
+          "application/json",
+        );
+        if (resBody.status === "ACCEPTED") {
+          return;
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+      throw "timed out pulling customer for ACCEPTED status";
+    }
+
+    function generateGetCustomerNetworkCall(
+      kycServerUrl: string,
+      customerId: string,
+      token: string,
+    ): NetworkCall {
+      const requestParamsObj: Record<string, string> = {
+        id: customerId,
+      };
+
+      const getCreateCustomerType = (config: Config): string | undefined =>
+        config?.sepConfig?.["12"]?.customers?.[
+          config?.sepConfig?.["12"]?.createCustomer
+        ].type;
+      const customerType = getCreateCustomerType(config);
+      if (customerType) requestParamsObj["type"] = customerType;
+      const searchParams = new URLSearchParams(requestParamsObj);
+
+      const getCustomerCall: NetworkCall = {
+        request: new Request(
+          kycServerUrl + `/customer?${searchParams.toString()}`,
+          {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          },
+        ),
+      };
+      return getCustomerCall;
+    }
+
+    const getSendingCustomerNetworkCall = generateGetCustomerNetworkCall(
+      this.context.expects.kycServerUrl,
+      this.context.expects.sendingCustomerId,
+      this.context.expects.sendingAnchorToken,
+    );
+
+    const getReceivingCustomerNetworkCall = generateGetCustomerNetworkCall(
+      this.context.expects.kycServerUrl,
+      this.context.expects.receivingCustomerId,
+      this.context.expects.sendingAnchorToken,
+    );
+
+    const customerPollingTimeout =
+      config.sepConfig["31"].customerPollingTimeout || 30;
+
+    try {
+      await pollCustomerAccepted(
+        getReceivingCustomerNetworkCall,
+        customerPollingTimeout,
+      );
+    } catch {
+      result.failure = makeFailure(this.failureModes.CUSTOMER_NOT_ACCEPTED, {
+        errors:
+          "timed out waiting for receiving customer to be in `ACCEPTED` status",
+      });
+      return result;
+    }
+
+    try {
+      await pollCustomerAccepted(
+        getSendingCustomerNetworkCall,
+        customerPollingTimeout,
+      );
+    } catch {
+      result.failure = makeFailure(this.failureModes.CUSTOMER_NOT_ACCEPTED, {
+        errors:
+          "timed out waiting for sending customer to be in `ACCEPTED` status",
+      });
+      return result;
+    }
```

Comment:
I think ideally, we poll these customers concurrently, otherwise we're actually giving the sending customer 2x `customerPollingTimeout`. My typescript is rusty right now but I know there is a way to do this, like Python's `await asyncio.gather(call(), call())`